File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        int k = atoi(argv[1]);
        printf("plaintext: ");
        string text = get_string();
        printf("ciphertext: ");
        for(int i=0, n=strlen(text); i<n; i++)
        {
            char ch = text[i];
            if (ch >= 'A' && ch <= 'Z') 
            {
                char index = (ch - 'A' + k) % 26;
                printf("%c", (char)(index + 'A'));
            } 
            else if (ch >= 'a' && ch <= 'z') 
            {
                char index = (ch - 'a' + k) % 26;
                printf("%c", (char)(index + 'a'));
            } 
            else {
                printf("%c", ch);
            }
        }
        printf("\n");
        return 0;
    }
    else
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}

File: caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("Usage: ./caesar k")
        return (1)
    else:
        key = int(sys.argv[1])
        print("plaintext: ", end="")
        text = cs50.get_string()
        print("ciphertext: ", end="")
        for c in text:
            if c.isalpha() and c.isupper():
                position = (ord(c) - 65 + key) % 26
                print(chr(position + 65), end="")
            elif c.isalpha and c.islower():
                position = (ord(c) - 97 + key) % 26
                print(chr(position + 97), end="")
            else:
                print(c, end="")
        print()
        return (0)

if __name__ == "__main__":
    main()

    
